src/components/views/BlackCardSelection.js

```diff
@@ -64,97 +60,78 @@ const BlackCardSelection = () => {
+                    setGameDuration(response.data.gameDuration)
+                    setTimeLeft(response.data.gameDuration - (new Date() - new Date(response.data.creationDate)))
+                    setIsCountdownActive(true)
+                })
+                .catch(error => {
+                    if(error.response.status === 404){
+                        setBlackCard(null);
+                        // && error.response.data.message === ""
+                    }else{
+                        console.error("Details:", error);
+                        alert("Invalid Input:\n " + handleError(error));
+                    }
+                })
+
+            api.get(`users/${userId}/games`)
+                .then(response => setCards(response.data))
+                .catch(error => {
                     console.error("Details:", error);
                     alert("Invalid Input:\n " + handleError(error));
-                }
-            }
-        }
-
-        // get card to choose from
-        async function fetchCards() {
-            try {
-                const response = await api.get(`users/${userId}/games`)
-                setCards(response.data)
-            }
-            catch (error) {
-                console.error("Details:", error);
-                alert("Invalid Input:\n " + handleError(error));
-            }
+                })
         }
-        fetchBlackCard();
-        fetchCards();
 
-    }, [countdown]); // when countdown is reached, new call to useEffects
+    }, [isCountdownActive]); // when countdown is reached, new call to useEffects
 
-    // stops the interval call
-    function checkIfIntervalStop() {
-        // the cals get stopped if the countdown is finished or the url is changed
-        if(countdown===true || !window.location.href.includes("/game/select/blackCard"))
-        {
-            clearInterval(timeInterval);
-        }
-    }
-
-    // gets called all seconds and updates the countdown
-    function calcTimeLeft(creationDate){
-        // difference between now and creationDate
-        const differenceInMs = new Date() - new Date(creationDate);
-        // calculate the time left
-        const timeLeft = ONE_DAY-differenceInMs;
-        // when there is no time left, the black cards get fetched again
-        if(timeLeft<=0){
-            // useEffect gets fetched again
-            setCountdown(true);
+    useEffect(() => {
+        if (isCountdownActive) {
+            const timer = setInterval(async () => {
+                const tL = calcTimeLeft()
+                setTimeLeft(tL)
+                if (tL <= 0) {
+                    setIsCountdownActive(false)
+                }
+            }, ONE_SECOND)
+            return () => clearInterval(timer)
         }
-        // checks if the function call stops
-        checkIfIntervalStop();
-        // set the time
-        setTimeLeft(timeLeft);
-    }
+    }, [isCountdownActive])
```

Comment:
both useEffect get called when isCountdownActive is changed. why two useEffects?
if, else isCountdownActive would be enough in one of them?

Reply:
Didn't work first, but seems to work now. Have updated.